src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+                    // builder.Append((object)<value>);
+                    // builder.Append(", "); // except for last member
+
+                    var member = printableMembers[i];
+                    block.Add(
+                        F.ExpressionStatement(F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(member.Name))));
+                    block.Add(
+                        F.ExpressionStatement(F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(" = "))));
+
+                    var value = member.Kind switch
+                    {
+                        SymbolKind.Field => F.Field(F.This(), (FieldSymbol)member),
+                        SymbolKind.Property => F.Property(F.This(), (PropertySymbol)member),
+                        _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                    };
+
+                    block.Add(F.ExpressionStatement(
+                        F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendObject),
+                            F.Convert(F.SpecialType(SpecialType.System_Object), value))));
+
+                    if (i < printableMembers.Length - 1)
+                    {
+                        block.Add(
+                            F.ExpressionStatement(F.Call(receiver: builder, F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(", "))));
+                    }
+                }
+
+                block.Add(F.Return(F.Literal(true)));
+
+                F.CloseMethod(F.Block(block.ToImmutableAndFree()));
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
+            {
+                diagnostics.Add(ex.Diagnostic);
+                F.CloseMethod(F.ThrowNull());
+            }
+        }
+
+        internal static MethodSymbol? FindValidPrintMethod(TypeSymbol containingType, CSharpCompilation compilation)
+        {
+            if (containingType.IsErrorType() || containingType.IsObjectType())
+            {
+                return null;
+            }
+
+            MethodSymbol? candidate = null;
+            var accessibility = (containingType.BaseTypeNoUseSiteDiagnostics.IsObjectType() && containingType.IsSealed) ?
+                Accessibility.Private :
+                Accessibility.Protected;
+
+            var stringBuilder = TypeWithAnnotations.Create(compilation.GetWellKnownType(WellKnownType.System_Text_StringBuilder));
+
+            foreach (var member in containingType.GetMembers(WellKnownMemberNames.PrintMembersMethodName))
+            {
+                if (member is MethodSymbol { DeclaredAccessibility: var foundAccessibility, IsStatic: false, ParameterCount: 1, Arity: 0, ParameterTypesWithAnnotations: var parameters } method &&
```

Comment:
Would it make sense to share the checks for the suitability of this candidate method with the checks for the suitability of a source PrintMembers method in SourceMemberContainerSymbol?

Reply:
Potentially. I'm following the model we established for clone method, so would leave this as-is for now.
`SourceMemberContainerSymbol` uses a signature-only symbol to find the one we want, which works because we have a data structure to help with that, whereas this one scans through members with a specific name and uses patterns to filter.

---
In reply to: [469414029](https://github.com/dotnet/roslyn/pull/46616#discussion_r469414029) [](ancestors = 469414029)